src/CMakeLists.txt

```diff
@@ -628,6 +581,9 @@ endif()
 
 add_library(common-objs OBJECT ${libcommon_files})
 set(ceph_common_objs
+  $<TARGET_OBJECTS:common-msg>
+  $<TARGET_OBJECTS:common-msg-async>
+  $<TARGET_OBJECTS:common-msg-simple>
```

Comment:
it's difficult to build the components like msg, auth src/common as static libraries as we have inter-dependencies between them.